core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1,78 +1,386 @@
+            .credentialsProvider(awsCredentialsProvider)
             .region(Region.of(config.getContactPoints().get(0)))
             .build();
+
+    ApplicationAutoScalingClientBuilder applicationAutoScalingClientBuilder =
+        ApplicationAutoScalingClient.builder();
+    config
+        .getEndpointOverride()
+        .ifPresent(e -> applicationAutoScalingClientBuilder.endpointOverride(URI.create(e)));
+    applicationAutoScalingClientBuilder.credentialsProvider(awsCredentialsProvider);
+    applicationAutoScalingClientBuilder.region(Region.of(config.getContactPoints().get(0)));
+    applicationAutoScalingClient = applicationAutoScalingClientBuilder.build();
+
     namespacePrefix = config.getNamespacePrefix();
     metadataManager = new DynamoTableMetadataManager(client, namespacePrefix);
   }
 
   @VisibleForTesting
   DynamoAdmin(DynamoTableMetadataManager metadataManager, Optional<String> namespacePrefix) {
     client = null;
+    applicationAutoScalingClient = null;
     this.metadataManager = metadataManager;
     this.namespacePrefix = namespacePrefix.map(n -> n + "_");
   }
 
+  @VisibleForTesting
+  DynamoAdmin(
+      DynamoDbClient dynamoDbClient,
+      ApplicationAutoScalingClient applicationAutoScalingClient,
+      DynamoTableMetadataManager metadataManager,
+      Optional<String> namespacePrefix) {
+    this.client = dynamoDbClient;
+    this.applicationAutoScalingClient = applicationAutoScalingClient;
+    this.metadataManager = metadataManager;
+    this.namespacePrefix = namespacePrefix;
+  }
+
   @Override
-  public void createNamespace(String namespace, Map<String, String> options)
-      throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+  public void createNamespace(String namespace, Map<String, String> options) {
+    // In Dynamo DB storage, namespace will be added to table name as prefix along with dot
+    // separator.
   }
 
   @Override
   public void createTable(
       String namespace, String table, TableMetadata metadata, Map<String, String> options)
       throws ExecutionException {
-    throw new UnsupportedOperationException("implement later");
+    CreateTableRequest.Builder requestBuilder = CreateTableRequest.builder();
+
+    List<AttributeDefinition> columnsToAttributeDefinitions = new ArrayList<>();
+    columnsToAttributeDefinitions.add(
+        AttributeDefinition.builder()
+            .attributeName(PARTITION_KEY)
+            .attributeType(ScalarAttributeType.S)
+            .build());
+    if (!metadata.getClusteringKeyNames().isEmpty()) {
+      columnsToAttributeDefinitions.add(
+          AttributeDefinition.builder()
+              .attributeName(CLUSTERING_KEY)
+              .attributeType(ScalarAttributeType.S)
+              .build());
+      for (String column : metadata.getClusteringKeyNames()) {
+        if (metadata.getColumnDataType(column) == DataType.BOOLEAN) {
```

Comment:
The code block in the for loop looks identical to the one below for the secondary index.
Can you unify it?